bundles/org.openhab.binding.mikrotik/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,431 @@
+			<channel id="txErrors" typeId="txErrors"/>
+			<channel id="rxErrors" typeId="rxErrors"/>
+			<!-- ethernet interface channels -->
+			<channel id="defaultName" typeId="defaultName"/>
+			<channel id="rate" typeId="ethernetRate"/>
+			<!-- cap interface channels -->
+			<channel id="state" typeId="state"/>
+			<channel id="registeredClients" typeId="registeredClients"/>
+			<channel id="authorizedClients" typeId="authorizedClients"/>
+			<!-- ppp & vpn interface channels -->
+			<channel id="uptime" typeId="uptime"/>
+			<channel id="upSince" typeId="upSince"/>
+		</channels>
+		<representation-property>name</representation-property>
+		<config-description>
+			<parameter name="name" type="text" required="true">
+				<label>Name</label>
+				<description>RouterOS Interface name (i.e. ether1)</description>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+	<thing-type id="wifiRegistration">
+		<supported-bridge-type-refs>
+			<bridge-type-ref id="routeros"/>
+		</supported-bridge-type-refs>
+		<label>RouterOS Wireless Client</label>
+		<description>A wireless client connected to a RouterOS wireless network (direct or CAPsMAN-managed)</description>
+		<channels>
+			<channel id="macAddress" typeId="macAddress"/>
+			<channel id="comment" typeId="comment"/>
+			<channel id="connected" typeId="connected"/>
+			<channel id="continuous" typeId="continuous"/>
+			<channel id="ssid" typeId="ssid"/>
+			<channel id="interface" typeId="interfaceName"/>
+			<channel id="signal" typeId="signal"/>
+			<channel id="uptime" typeId="uptime"/>
+			<channel id="upSince" typeId="upSince"/>
+			<channel id="lastSeen" typeId="lastSeen"/>
+			<channel id="txRate" typeId="txRate"/>
+			<channel id="rxRate" typeId="rxRate"/>
+			<channel id="txPacketRate" typeId="txPacketRate"/>
+			<channel id="rxPacketRate" typeId="rxPacketRate"/>
+			<channel id="txBytes" typeId="txBytes"/>
+			<channel id="rxBytes" typeId="rxBytes"/>
+			<channel id="txPackets" typeId="txPackets"/>
+			<channel id="rxPackets" typeId="rxPackets"/>
+		</channels>
+		<representation-property>macAddress</representation-property>
+		<config-description>
+			<parameter name="mac" type="text" required="true">
+				<label>Client MAC</label>
+				<description>WiFi client MAC address</description>
+			</parameter>
+			<parameter name="ssid" type="text" required="false">
+				<label>SSID</label>
+				<description>
+					Constraining SSID for the WiFi client (optional). If client will connect to another SSID,
+					this thing
+					will stay offline until client reconnects to specified SSID.
+				</description>
+			</parameter>
+			<parameter name="considerContinuous" type="integer" required="false">
+				<label>Consider Home Interval</label>
+				<description>The interval in seconds to treat the client as connected permanently</description>
+				<default>180</default>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+
+	<channel-type id="freeSpace">
+		<item-type>Number</item-type>
```

Comment:
This comment applies to all channels.
If you implement UOM then you get openHAB able to produce graphs right away when you create the equipment in OH3.

See https://www.openhab.org/docs/concepts/units-of-measurement.html
The item-type would change to...
Number:DataAmount

Use the other bindings as examples of how to implement this.

Reply:
Thanks for the advice! I'll look into this.